blueprints/geometry/operations.py

```diff
@@ -0,0 +1,83 @@
+"""Geometry operations module."""
+
+import math
+from enum import Enum, auto
+
+from shapely import Point
+
+from blueprints.type_alias import RAD
+
+
+class CoordinateSystemOptions(Enum):
+    """Enum of the coordinate system options."""
+
+    XY = auto()
+    XZ = auto()
+    YZ = auto()
+
+
+def calculate_rotation_angle(
+    start_point: Point,
+    end_point: Point,
+    coordinate_system: CoordinateSystemOptions = CoordinateSystemOptions.XY,
+) -> RAD:
+    """
+    Calculates rotation of an end point in relation to the start point in a given plane/coordinate system [rad].
+
+    - Start point lies in the origin center of the quadrant.
+    - The rotation is calculated in the given plane in relation to the start point.
+    - The rotation is calculated in the range [0, 2*pi].
+    - The rotation is calculated in the counter-clockwise direction.
+
+    Parameters
+    ----------
+    start_point: Point
+        Starting point that will be used as reference.
+    end_point: Point
+        End point
+    coordinate_system: CoordinateSystemOptions
+        Desired plane that will be used as reference to calculate the rotation. Standard is XY-plane.
+
+    Returns
+    -------
+    RAD
+        rotation of end point relative to begin node in radians in the given plane [rad].
+
+    Raises
+    ------
+    ValueError
+        If start_point and end_point are the same.
+        If start_point or end_point do not have z value when rotation angle in XZ or YZ plane is requested.
+    """
+    if list(start_point.coords) == list(end_point.coords):
+        msg = f"Start and end point can't be equal. start={start_point} | end={end_point}"
+        raise ValueError(msg)
+
+    if coordinate_system != CoordinateSystemOptions.XY and (not start_point.has_z or not end_point.has_z):
+        msg = f"Coordinate system '{coordinate_system}' requires z value in both points."
+        raise ValueError(msg)
+
+    # Map coordinate system to the corresponding attributes
+    coordinates_map = {
+        CoordinateSystemOptions.XY: ("x", "y"),
+        CoordinateSystemOptions.XZ: ("x", "z"),
+        CoordinateSystemOptions.YZ: ("y", "z"),
+    }
```

Comment:
Could this have been done in de CoordinateSystemOptions Enum?
Suggestion: You could use "xy" in the Enum and then use split here.